ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLog.java

```diff
@@ -382,26 +385,29 @@ protected CompletableFuture<Long> appendEntryImpl(LogEntryProto entry) {
       final Timekeeper.Context appendEntryTimerContext = getRaftLogMetrics().startAppendEntryTimer();
       validateLogEntry(entry);
       final LogSegment currentOpenSegment = cache.getOpenSegment();
+      boolean rollOpenSegment = false;
```

Comment:
do we need to check here when `addOpenSegment`?

Reply:
Looking at the contents of shouldEvict, it seems that a check is not necessary because nothing will change.

```Java
    final CacheInfo closedSegmentsCacheInfo = closedSegments.getCacheInfo();
    if (closedSegmentsCacheInfo.getCount() > maxCachedSegments) {
      return true;
    }

    final long size = closedSegmentsCacheInfo.getSize()
        + Optional.ofNullable(openSegment).map(LogSegment::getTotalCacheSize).orElse(0L);
    return size > maxSegmentCacheSize;
  }
```

Reply:
For `rollOpenSegment`, we will close the open segement so it increases the count.  For `addOpenSegment`, nothing will change as you mentioned.